src/components/TextInput/BaseTextInput.js

```diff
@@ -21,381 +21,380 @@ import isInputAutoFilled from '../../libs/isInputAutoFilled';
 import * as Pressables from '../Pressable';
 
 const PressableWithoutFeedback = Pressables.PressableWithoutFeedback;
-class BaseTextInput extends Component {
-    constructor(props) {
-        super(props);
-
-        const value = props.value || props.defaultValue || '';
-        const activeLabel = props.forceActiveLabel || value.length > 0 || Boolean(props.prefixCharacter);
-
-        this.state = {
-            isFocused: false,
-            labelTranslateY: new Animated.Value(activeLabel ? styleConst.ACTIVE_LABEL_TRANSLATE_Y : styleConst.INACTIVE_LABEL_TRANSLATE_Y),
-            labelScale: new Animated.Value(activeLabel ? styleConst.ACTIVE_LABEL_SCALE : styleConst.INACTIVE_LABEL_SCALE),
-            passwordHidden: props.secureTextEntry,
-            textInputWidth: 0,
-            prefixWidth: 0,
-            selection: props.selection,
-            height: variables.componentSizeLarge,
-
-            // Value should be kept in state for the autoGrow feature to work - https://github.com/Expensify/App/pull/8232#issuecomment-1077282006
-            value,
-        };
 
-        this.input = null;
-        this.isLabelActive = activeLabel;
-        this.onPress = this.onPress.bind(this);
-        this.onFocus = this.onFocus.bind(this);
-        this.onBlur = this.onBlur.bind(this);
-        this.setValue = this.setValue.bind(this);
-        this.togglePasswordVisibility = this.togglePasswordVisibility.bind(this);
-        this.dismissKeyboardWhenBackgrounded = this.dismissKeyboardWhenBackgrounded.bind(this);
-        this.storePrefixLayoutDimensions = this.storePrefixLayoutDimensions.bind(this);
+function dismissKeyboardWhenBackgrounded(nextAppState) {
+    if (!nextAppState.match(/inactive|background/)) {
+        return;
     }
 
-    componentDidMount() {
-        if (this.props.disableKeyboard) {
-            this.appStateSubscription = AppState.addEventListener('change', this.dismissKeyboardWhenBackgrounded);
+    Keyboard.dismiss();
+}
+
+function BaseTextInput(props) {
```

Comment:
based on this PR https://github.com/Expensify/eslint-config-expensify/pull/64 we can use destructuring props - which will clean code a little bit

Reply:
in current situation - i think it will be good idea to move away from defining every time a new variable like that:
    const propIsDisabled = props.disabled;
    const propOnPress = props.onPress;

Reply:
I think we should still avoid destructuring props https://expensify.slack.com/archives/C01GTK53T8Q/p1686062674061049?thread_ts=1685727880.152119&cid=C01GTK53T8Q

https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#destructuring

Reply:
Okay, I double checked with the documentation and removed the `useCallbacks` where we don't need them according to the docs. This removed the weird prop usage anyway!